examples/computer_vision/fashion_product_images/scripts/3-train-model.py

```diff
@@ -1,82 +1,13 @@
-import numpy as np
-import torch
-from torch import nn, optim
 from torch.utils.data import DataLoader
-from torchvision.transforms import v2
-from tqdm import tqdm
 
 from datachain import DataChain
 from datachain.lib.pytorch import label_to_int
 
-transform = v2.Compose(
-    [
-        v2.Compose([v2.ToImage(), v2.ToDtype(torch.float32, scale=True)]),
-        v2.Resize((64, 64)),
-        v2.RGB(),
-    ]
-)
-
-
-class CNN(nn.Module):
-    def __init__(self, num_classes):
-        super().__init__()
-        self.conv1 = nn.Conv2d(3, 16, kernel_size=3, stride=2, padding=1)
-        self.conv2 = nn.Conv2d(16, 32, kernel_size=3, stride=2, padding=1)
-        self.conv3 = nn.Conv2d(32, 64, kernel_size=3, stride=2, padding=1)
-        self.fc1 = nn.Linear(64 * 8 * 8, 512)
-        self.fc2 = nn.Linear(512, num_classes)
-
-    def forward(self, x):
-        x = torch.relu(self.conv1(x))
-        x = torch.relu(self.conv2(x))
-        x = torch.relu(self.conv3(x))
-        x = x.view(-1, 64 * 8 * 8)
-        x = torch.relu(self.fc1(x))
-        return self.fc2(x)
-
-
-#### Train  #####
-def train_model(train_loader, num_classes, num_epochs=20, lr=0.001):
-    """
-    Trains a convolutional neural network on a given dataset.
-
-    Args:
-        train_loader (DataLoader): DataLoader for the training data.
-        num_classes (int): Number of classes in the dataset.
-        num_epochs (int, optional): Number of epochs to train the model. Defaults to 20.
-        lr (float, optional): Learning rate for the optimizer. Defaults to 0.001.
-
-    Returns:
-        tuple: A tuple containing the trained model and the optimizer used for training.
-
-    """
+# For Local Development
+from src.train import train_model, transform
```

Comment:
Have you run the script locally?

Reply:
yes, at least to test it works.

Reply:
I get this: 

```
❯ python examples/computer_vision/fashion_product_images/scripts/3-train-model.py
Traceback (most recent call last):
  File "/datachain/examples/computer_vision/fashion_product_images/scripts/3-train-model.py", line 7, in <module>
    from src.train import train_model, transform
ModuleNotFoundError: No module named 'src'
```

Reply:
Ok, I got it!  To test, I run scripts from inside the `fashion_product_images/` directory. 
I don't expect users will run scripts outside the Studio.

Reply:
We want to keep all the scripts working and usable for anyone who stumbles across them. 

I am working on adding all of the scripts to a test suite that will run nightly. If we use `nbconvert` to create an exact (script) copy of each notebook we can verify that all steps run without error.

My $0.02 is that the duplication isn't that bad and having all of the code exposed and visible (as opposed to hidden) would be a better idea for tutorial purposes. This also keeps the instructions for running things in Studio simple(r). For example, we have copies of the same `CNN` across different scripts already (`examples/get_started/torch-loader.py` / `examples/computer_vision/fashion_product_images/src/train.py`.) and it doesn't seem to be a problem.

Reply:
@mattseddon, let's sync on this testing. May be a short call will be faster. 
I'm also have a solution for nightly tests, but with `papermill` :))) I'll share with you, let's discuss then

Reply:
I have an additional requirement here - I need to keep Studio examples somewhere and make it as simple as possible to copy-paste them into Studio (along with additional imported modules). 
- that's why we have `scripts` in `fashion...` examples dir 
- they are in the Studio format 
- the `train.py` is designed intentionally to be attached to Studio 
- to test them I use `datachain query ...` CLI, not `python  ... `

Reply:
no worries, you can maintain the examples. I'll close my PRs

Reply:
Please tag me in the papermill PR. I'm interested to see how that works.